extensions/wikia/Blogs/BlogArticle.php

```diff
@@ -135,66 +145,101 @@ private function showBlogListing() {
 	 * @access public
 	 */
 	public function clearBlogListing() {
-		global $wgRequest, $wgMemc, $wgLang;
+		global $wgMemc;
 
 		// Clear Oasis rail module
-		$mcKey = wfMemcKey( "OasisPopularBlogPosts", $wgLang->getCode() );
+		$mcKey = $this->blogListingOasisMemcacheKey();
 		$wgMemc->delete( $mcKey );
 
-		$user = $this->mTitle->getPrefixedDBkey();
-		foreach ( range( 0, 5 ) as $page ) {
-			$wgMemc->delete( $this->blogListingMemcacheKey( $user, $page ) );
```

Comment:
The previous cache clearing above was totally arbitrary, clearing out a hard coded 5 pages of blog listings and then calling it a day.  Below we get the actual count and clear those.